subsystem/kubeapi_test.go

```diff
@@ -1821,6 +1823,64 @@ spec:
 			return len(getClusterFromDB(ctx, kubeClient, db, clusterKey, waitForReconcileTimeout).Hosts)
 		}, "2m", "2s").Should(Equal(0))
 	})
+
+	It("Verify garbage collector inactive cluster deregistration isn't triggered for clusters created via ClusterDeployment", func() {
+		deployClusterDeploymentCRD(ctx, kubeClient, clusterDeploymentSpec)
+		deployInfraEnvCRD(ctx, kubeClient, infraNsName.Name, infraEnvSpec)
+		deployAgentClusterInstallCRD(ctx, kubeClient, aciSpec, clusterDeploymentSpec.ClusterInstallRef.Name)
+		clusterKey := types.NamespacedName{
+			Namespace: Options.Namespace,
+			Name:      clusterDeploymentSpec.ClusterName,
+		}
+
+		By("Update cluster's updated_at attribute to become eligible for deregistration due to inactivity")
+		cluster := getClusterFromDB(ctx, kubeClient, db, clusterKey, waitForReconcileTimeout)
+		db.Model(&cluster).UpdateColumn("updated_at", time.Now().AddDate(-1, 0, 0))
+
+		By("Verify no event for deregistering the cluster was emitted")
+		msg := "Cluster is deregistered"
+		Consistently(func() []*common.Event {
```

Comment:
why check consistency?

Reply:
To make sure that during that period the execution of the garbageCollector.DeregisterClusterInternal did not occur and as a result the cluster was not deregistered.